internal/factories/factories.go

```diff
@@ -56,6 +56,11 @@ type ReceiverFactory interface {
 		consumer consumer.MetricsConsumer) (receiver.MetricsReceiver, error)
 }
 
+// ErrDataTypeIsNotSupported can be returned by CreateTraceReceiver or
```

Comment:
nit: it will be good if the message states what is the telemetry type not supported ("metrics", "traces", "logs"). 

Related q: I assume it is public so each factory can use it but is it expected that anyone will handle this error?

Reply:
I agree, I left it out because it was not yet clear to me how exactly the error message will be used. I will add it as soon as it is clear that the message is going to be visible to the end user (there is a chance the  the error message for end user will be formatted elsewhere and we will not need it here, so that's why I am delaying).

Reply:
sgtm